src/views/manage-identities/ManageIdentities.tsx

```diff
@@ -0,0 +1,29 @@
+import { useContext, useEffect } from "react";
+import { WatchContractEventReturnType } from "viem";
+import { watchTrexFactoryTrexSuiteDeployedEvent } from "@/services/contracts/wagmiGenActions";
+import { Eip3643Context } from "@/contexts/Eip3643Context";
+import { useWalletInterface } from "@/services/wallets/useWalletInterface";
+import { ManageTokensList } from "@/components/ManageIdentitiesList";
+
+export default function ManageIdentities() {
+  const { setDeployedTokens } = useContext(Eip3643Context);
+  const { accountId } = useWalletInterface();
+
+  useEffect(() => {
+    const unsubTokens: WatchContractEventReturnType =
+      watchTrexFactoryTrexSuiteDeployedEvent({
```

Comment:
we already have this listener in parent `src/views/eip3643/EIP3643.tsx`, I'd suggest making managing identities as a sub page of that to improve the structure and reduce repeated calls